v2/runner/collector_int_test.go

```diff
@@ -1095,3 +1095,105 @@ func (s *SuiteT) TestCollectingOTELTypes() {
+	ctx.DefineMetric("/kubernetes/pod/[node]/[namespace]/[pod]/status/phase/Failed", "", true, "all containers in the pod have terminated, and at least one container has terminated in a failure")
+	ctx.DefineMetric("/kubernetes/pod/[node]/[namespace]/[pod]/status/phase/Unknown", "", true, "for some reason the state of the pod could not be obtained, typically due to an error in communicating with the host of the pod")
+	ctx.DefineMetric("/kubernetes/pod/[node]/[namespace]/[pod]/status/condition/ready", "", false, "specifies if the pod is ready to serve requests")
+	ctx.DefineMetric("/kubernetes/pod/[node]/[namespace]/[pod]/status/condition/scheduled", "", false, "status of the scheduling process for the pod")
+
+	ctx.DefineMetric("/rabbitmq/queue/count", "", false, "number of defined queues")
+
+	return nil
+}
+
+func (c *collectorWithGlobalPrefix) Collect(ctx plugin.CollectContext) error {
+	Convey("Validate AddMetric", c.t, func() {
+		So(ctx.AddMetric("/kubernetes/pod/node1/ns1/pod1/status/phase/Pending", 12), ShouldBeNil)
+		So(ctx.AddMetric("/kubernetes/pod/node1/ns1/pod1/status/phase/Succeeded", 11), ShouldBeNil)
+		So(ctx.AddMetric("/kubernetes/pod/node1/ns1/pod1/status/condition/ready", 7), ShouldBeNil)
+		So(ctx.AddMetric("/rabbitmq/queue/count", 15), ShouldBeNil)
+		So(ctx.AddMetric("/kubernetes/pod/node1/ns1/pod1/status/condition/scheduled", 18), ShouldBeNil)
+	})
+
+	return nil
+}
+
+func (s *SuiteT) TestCollectorWithGlobalPrefix_RemoveFromOutput() {
+	// Arrange
+	jsonConfig := []byte(`{}`)
+	mtsSelector := []string{
+		"/kubernetes/pod/node1/*/*/status/phase/Pending",
+		"/kubernetes/pod/node1/*/*/status/phase/Succeeded",
+		"/kubernetes/pod/node1/ns1/pod1/status/condition/ready",
+		"/rabbitmq/**",
+	}
+
+	collector := &collectorWithGlobalPrefix{
+		t:                s.T(),
+		prefixName:       "/swi",
+		removeFromOutput: true,
+	}
+
+	ln := s.startCollector(collector)
+	s.startClient(ln.Addr().String())
+
+	Convey("Validate collector can define metrics with different namespace root when globalPrefix is set", s.T(), func() {
+		_, _ = s.sendLoad("task-1", jsonConfig, mtsSelector)
+
+		mts, err := s.sendCollect("task-1")
+		So(err, ShouldBeNil)
+		So(len(mts.MetricSet), ShouldEqual, 4) // /kubernetes/pod/node1/ns1/pod1/status/condition/scheduled is filtered out
+
+		So(mts.MetricSet[0].Namespace[0].Value, ShouldEqual, "kubernetes")
+		So(mts.MetricSet[1].Namespace[0].Value, ShouldEqual, "kubernetes")
+		So(mts.MetricSet[2].Namespace[0].Value, ShouldEqual, "kubernetes")
+		So(mts.MetricSet[3].Namespace[0].Value, ShouldEqual, "rabbitmq")
+	})
+}
+
+func (s *SuiteT) TestCollectorWithGlobalPrefix_LeaveInOutput() {
+	// Arrange
+	jsonConfig := []byte(`{}`)
+	var mtsSelector []string
+
+	collector := &collectorWithGlobalPrefix{
+		t:                s.T(),
+		prefixName:       "/swi",
```

Comment:
Then every plugin that is bundled into swi has to use the same separator char? So we have to unify elastic, haproxy, k8s and docker

Reply:
no, this is only example test, 
global prefix allows adding prefix "#swi"